api/lib/domain/events/handle-badge-acquisition.js

```diff
@@ -7,15 +9,18 @@ const handleBadgeAcquisition = async function({
   campaignParticipationResultRepository,
 }) {
   if (completedAssessmentBelongsToACampaign(assessmentCompletedEvent)) {
-    const badge = await fetchPossibleCampaignAssociatedBadge(assessmentCompletedEvent, badgeRepository);
-    if (isABadgeAssociatedToCampaign(badge)) {
-      const campaignParticipationResult = await fetchCampaignParticipationResults(assessmentCompletedEvent, badge, campaignParticipationResultRepository);
-      if (isBadgeAcquired(campaignParticipationResult, badge.badgeCriteria, badgeCriteriaService)) {
-        await badgeAcquisitionRepository.create({
-          badgeId: badge.id,
-          userId: assessmentCompletedEvent.userId
-        }, domainTransaction);
-      }
+    const campaignBadges = await fetchPossibleCampaignAssociatedBadges(assessmentCompletedEvent, badgeRepository);
+    const campaignParticipationResult = await fetchCampaignParticipationResults(assessmentCompletedEvent, campaignBadges, campaignParticipationResultRepository);
+
+    const badgesBeingAcquired = campaignBadges.filter((badge) => isBadgeAcquired(campaignParticipationResult, badge, badgeCriteriaService));
+    const badgesAcquisitionToCreate = badgesBeingAcquired.map((badge) => {
```

Comment:
À ce moment, on a les badges déjà acquis et non en cours d'acquisition. On cherche effectivement à matérialiser leur acquisition en base de données . Je verrais plutôt

```suggestion
    const badgesAcquired = campaignBadges.filter((badge) => isBadgeAcquired(campaignParticipationResult, badge, badgeCriteriaService));
    const badgesAcquisitionToCreate = badgesAcquired.map((badge) => {
```

Reply:
Je suis mitigée sur la terminologie =/ Les badges sont effectivement considérés acquis, mais en base de données, ils ne le sont pas encore car pas encore insérés dans la table `badge-acquisitions`.
Comme ils ne sont pas encore "officiellement" acquis, ils sont dans l'état "beingAcquired."